ccmlib/scylla_repository.py

```diff
@@ -414,25 +409,30 @@ def download_packages(version_dir, packages, s3_url, scylla_product, version, ve
     return package_version, packages
 
 
-def setup_scylla_manager(scylla_manager_package=None):
+def setup_scylla_manager(scylla_manager_package=None, verbose=False):
 
     """
     download and cache scylla-manager RPMs,
     :return:
     """
 
     if scylla_manager_package and '--scylla-manager':
-        m = hashlib.md5()
-        m.update(scylla_manager_package.encode('utf-8'))
+        dir_hash = get_url_hash(scylla_manager_package)
 
-        # select a dir to change this version of scylla-manager based on the md5 of the path
-        manager_install_dir = directory_name(os.path.join('manager', m.hexdigest()))
+        # select a dir to change this version of scylla-manager based on the md5 of the path or the etag of the s3 object
+        manager_install_dir = directory_name(os.path.join('manager', dir_hash))
         if not os.path.exists(manager_install_dir):
             os.makedirs(manager_install_dir)
-            tar_data = requests.get(scylla_manager_package, stream=True)
             destination_file = os.path.join(manager_install_dir, "manager.tar.gz")
-            with open(destination_file, mode="wb") as f:
-                f.write(tar_data.raw.read())
+
+            if os.path.exists(scylla_manager_package) and scylla_manager_package.endswith('.tar.gz'):
+                destination_file = scylla_manager_package
+            elif is_valid(scylla_manager_package):
+                _, target = tempfile.mkstemp(suffix=".tar.gz", prefix="ccm-")
```

Comment:
what is `target` for? presumably, you wanted to use it for the path for the temporary file for the compressed scylla manager package. but it turns out you actually download the file to `destination_file`.

Reply:
you are correct it's a copy paste, and this line should replace 
```
destination_file = os.path.join(manager_install_dir, "manager.tar.gz")
```

name should be temporary

Reply:
move the temp file into the proper place